packages/core/src/MatchMaker.ts

```diff
@@ -165,8 +165,8 @@ export async function joinById(roomId: string, clientOptions: ClientOptions = {}
 /**
  * Perform a query for all cached rooms
  */
-export async function query(conditions: Partial<IRoomListingData> = {}) {
-  return await driver.find(conditions);
+export function query(conditions: Partial<IRoomListingData> = {}) {
```

Comment:
this is a good improvement to make. The implementation is much better without extra layers of promises